### ISSUE TITLE
feat: support tracking consensus updates in helios-ts 

### ISSUE THREAD

**Issue by rdvorkin** [a16z/helios#302]
Support in some way tracking and getting notified when a new update is applied. The usecase I want to implement is setting a hook on new optimistic update, to do some action (For example, re-check a list of balances) when the update is applied.

Implementing something like https://nodejs.org/api/events.html#emitteroneventname-listener would be ideal, however even a function returning the current status (latest update applied, current block etc) would be great.

**Comment by ncitron** [comment]
Would something that mimics the websocket block subscription api make sense here potentially?

**Comment by rdvorkin** [comment]
@ncitron Yeah I suppose so, could you link an example of how that would look like?

**Comment by rdvorkin** [comment]
@ncitron Yes this seems perfect!

**Comment by ppca** [comment]
Hey I'm wondering if there's support for `eth_subscribe` in the rust crate now?

**Comment by ppca** [comment]
Thanks @karen-sarkisyan !

**Comment by ncitron** [comment]
@karen-sarkisyan Is this for the eth_subscribe newHeads or the filter one? We already support tracking new heads for a filter, but have no subscribe support so if thats it feel free.

**Comment by ppca** [comment]
@ncitron What does `tracking new heads for a filter` mean?
What I want to do is basically 
```
web3_ws.eth_subscribe().subscribe_logs(filter)
```
Do you already have support to enable this?
